core/src/test/java/org/bitcoinj/crypto/HDKeyDerivationTest.java

```diff
@@ -100,7 +99,30 @@ public void testDeriveFromPublicParent() {
         assertTrue(fromPublicWithInversion.isPubKeyOnly());
         assertFalse(fromPublicWithInversion.isEncrypted());
     }
+    @Test (expected = java.lang.IllegalArgumentException.class)
```

Comment:
I think these two new tests don't work as expected. They actually throw IllegalArgumentException due to a different reason than expected, and earlier than expected (which I understand is the last line of each test).